vms/platformvm/txs/builder/builder.go

```diff
@@ -426,334 +162,192 @@ func (b *builder) NewTransformSubnetTx(
 
-	vdr := &txs.SubnetValidator{
-		Validator: txs.Validator{
-			NodeID: nodeID,
-			Start:  startTime,
-			End:    endTime,
-			Wght:   stakeAmount,
-		},
-		Subnet: constants.PrimaryNetworkID,
-	}
-
-	rewardOwner := &secp256k1fx.OutputOwners{
-		Threshold: 1,
-		Addrs:     []ids.ShortID{rewardAddress},
-	}
-
 	utx, err := pBuilder.NewAddPermissionlessDelegatorTx(
 		vdr,
-		b.ctx.AVAXAssetID,
-		rewardOwner,
-		options(changeAddr, memo)...,
+		assetID,
+		rewardsOwner,
+		options...,
 	)
 	if err != nil {
 		return nil, fmt.Errorf("failed building add permissionless delegator tx: %w", err)
 	}
 
-	tx, err := walletsigner.SignUnsigned(context.Background(), pSigner, utx)
-	if err != nil {
-		return nil, err
-	}
-	return tx, tx.SyntacticVerify(b.ctx)
+	return walletsigner.SignUnsigned(context.Background(), pSigner, utx)
 }
 
-func (b *builder) NewAddSubnetValidatorTx(
-	weight,
-	startTime,
-	endTime uint64,
-	nodeID ids.NodeID,
-	subnetID ids.ID,
+func (b *Builder) NewAddSubnetValidatorTx(
+	vdr *txs.SubnetValidator,
 	keys []*secp256k1.PrivateKey,
-	changeAddr ids.ShortID,
-	memo []byte,
+	options ...common.Option,
 ) (*txs.Tx, error) {
 	pBuilder, pSigner := b.builders(keys)
 
-	vdr := &txs.SubnetValidator{
-		Validator: txs.Validator{
-			NodeID: nodeID,
-			Start:  startTime,
-			End:    endTime,
-			Wght:   weight,
-		},
-		Subnet: subnetID,
-	}
-
 	utx, err := pBuilder.NewAddSubnetValidatorTx(
 		vdr,
-		options(changeAddr, memo)...,
+		options...,
 	)
 	if err != nil {
 		return nil, fmt.Errorf("failed building add subnet validator tx: %w", err)
 	}
 
-	tx, err := walletsigner.SignUnsigned(context.Background(), pSigner, utx)
-	if err != nil {
-		return nil, err
-	}
-	return tx, tx.SyntacticVerify(b.ctx)
+	return walletsigner.SignUnsigned(context.Background(), pSigner, utx)
 }
 
-func (b *builder) NewRemoveSubnetValidatorTx(
+func (b *Builder) NewRemoveSubnetValidatorTx(
 	nodeID ids.NodeID,
 	subnetID ids.ID,
 	keys []*secp256k1.PrivateKey,
-	changeAddr ids.ShortID,
-	memo []byte,
+	options ...common.Option,
 ) (*txs.Tx, error) {
 	pBuilder, pSigner := b.builders(keys)
 
 	utx, err := pBuilder.NewRemoveSubnetValidatorTx(
 		nodeID,
 		subnetID,
-		options(changeAddr, memo)...,
+		options...,
 	)
 	if err != nil {
 		return nil, fmt.Errorf("failed building remove subnet validator tx: %w", err)
 	}
 
-	tx, err := walletsigner.SignUnsigned(context.Background(), pSigner, utx)
-	if err != nil {
-		return nil, err
-	}
-	return tx, tx.SyntacticVerify(b.ctx)
+	return walletsigner.SignUnsigned(context.Background(), pSigner, utx)
 }
 
-func (b *builder) NewTransferSubnetOwnershipTx(
+func (b *Builder) NewTransferSubnetOwnershipTx(
 	subnetID ids.ID,
-	threshold uint32,
-	ownerAddrs []ids.ShortID,
+	owner *secp256k1fx.OutputOwners,
 	keys []*secp256k1.PrivateKey,
-	changeAddr ids.ShortID,
-	memo []byte,
+	options ...common.Option,
 ) (*txs.Tx, error) {
 	pBuilder, pSigner := b.builders(keys)
 
-	utils.Sort(ownerAddrs) // sort control addresses
```

Comment:
this sort is carried out in the wallet right?

Reply:
https://github.com/ava-labs/avalanchego/blob/p-chain_spend_cleanup_more_nits/wallet/chain/p/builder/builder.go#L572